### PR TITLE
Major refactor and consistency fixes

### PR DESCRIPTION
**Pull request by dennlinger**
This PR addresses several issues that have come up in the discussions:

* Addresses #2 , by simply removing outdated code snippets that caused an issue with specific (newer) `transformers` versions. Note that this can be further improved in the future by revisiting how much of our actual code is required, and whether we can substitute these classes by inheriting from `transformers` directly, thus avoiding code duplication.
* Adds the option to **install the repository as a pip package**. This solves several (related) issues, mostly centered around correct imports from parent-level folders, which is non-trivial with Python. For now, the option is to install it with `python3 -m pip install .` (in the root folder), however, I am actively working to get it on PyPI.org as well. This way, I have also verified that the examples are finally running correctly, which was a previous issue.
* I've started to be more serious about "separation of concerns", and hopefully improved readability of the actual module by moving our scripts into subfolders.
* Renaming some of the classes to be more consistent (e.g., `BERT_CRF_NER` is now `BertWithCRF`), as well as more expressive (`NumbertTokenizer` is now `DateTokenizer`).
* I've contained the refactor of actual code to a minimum for now, since I was first trying to get the right code structure, which would then allow us to improve other aspects in a separate PRs, without further crowding changes into this one. There are some obvious fixes, but no deeper changes to the "business logic".
* Also includes re-zipped versions of the LFS files (data and results), since the MacOS-zipped folders contain "ghost files" which show up on Linux systems, and cause the folder structure to become invalid.

**Disclaimer:** I have only tested functionality on the two model_hub scripts, which is not ideal. If you have any suggestions for which tests/scripts to run to see whether something is broken, please let me know.  
This is also relevant for *paths referenced in files*. While I believe to have taken care of most paths, the package change might have invalidated some paths that were not found by PyCharm, so no guarantees here.

### CODE REVIEW

File: data_preparation/heideltime/heideltime_generate_tempeval_data.py [hunk 1]
```diff
@@ -54,11 +55,11 @@ def process_file(in_fp, heideltime_parser, tweets):
     text, annotations, date = get_text_and_annotations_and_date(in_fp)
     heideltime_parser.set_document_time(date)
     heidel_xml = heideltime_parser.parse(text)
-    heidel_xml = removeHeaderAndFooterFromHeidelTimedDoc(heidel_xml)
+    heidel_xml = remove_header_and_footer_from_heideltime_doc(heidel_xml)
 
     with open(in_fp) as f:
         all_line = "\n".join(f.readlines())
-    if not tweets:#tweets have a slightly new line format
+    if not tweets:  # tweets has a slightly dfferent line format
```
Comment (dennlinger) [inline data_preparation/heideltime/heideltime_generate_tempeval_data.py:62]: Small spelling mistake.

File: data_preparation/heideltime/subset_heideltime_data.py [hunk 2]
```diff
@@ -20,67 +21,66 @@ def create_new_files(file, out_output, total_sample, total_date, total_no_date):
...
-                        short_examples=short_examples+1
-                    if   len(j_line["tagged_text"]) > 3000:
-                        long_examples=long_examples+1
+                    if len(json_line["tagged_text"]) < 50:
+                        short_examples = short_examples + 1
+                    if len(json_line["tagged_text"]) > 3000:
+                        long_examples = long_examples + 1
 
-    print("too short examples<50:",short_examples)
+    print("too short examples<50:", short_examples)
```
Comment (dennlinger) [inline data_preparation/heideltime/subset_heideltime_data.py:65]: Spelling, for future reference.

**Review by dennlinger — COMMENTED** [review]
Incomplete review, but checked some files myself.

**Review by satya77 — APPROVED** [review]

**Comment by satya77** [comment]
At the moment the pip installation only works with "pip install .".

**Pull request merged**